test/serializers/associations_test.rb

```diff
@@ -125,6 +125,88 @@ def test_associations_custom_keys
         assert expected_association_keys.include? :writer
         assert expected_association_keys.include? :site
       end
+
+      class NamespacedResourcesTest < Minitest::Test
+        class ResourceNamespace
+          Post    = Class.new(::Model)
+          Comment = Class.new(::Model)
+          Author  = Class.new(::Model)
+          Description = Class.new(::Model)
+          class PostSerializer < ActiveModel::Serializer
+            has_many :comments
+            belongs_to :author
+            has_one :description
+          end
+          CommentSerializer     = Class.new(ActiveModel::Serializer)
+          AuthorSerializer      = Class.new(ActiveModel::Serializer)
+          DescriptionSerializer = Class.new(ActiveModel::Serializer)
+        end
+
+        def setup
+          @comment = ResourceNamespace::Comment.new
+          @author = ResourceNamespace::Author.new
+          @description = ResourceNamespace::Description.new
+          @post = ResourceNamespace::Post.new(comments: [@comment],
+                                              author: @author,
+                                              description: @description)
+          @post_serializer = ResourceNamespace::PostSerializer.new(@post)
```

Comment:
So the previous behavior was not to look for a serializer in the global namespace unless the model itself is in the global namespace, which this PR preserves.